File: internal/core/entity/item.go
package entity

import (
	"gorm.io/gorm"
)

type Item struct {
	Name        string `gorm:"uniqueIndex:idx_name"`
	Description string
	AvailableAmount   int
	TotalAmount int    `gorm:"not null;default:0"`
	Status      string `gorm:"type:VARCHAR(20);default:'available'"` // available, borrowed, maintenance, lost
	gorm.Model
}

File: frontend/src/types/item.ts
export type ItemStatus = 'available' | 'borrowed' | 'maintenance' | 'lost'

export interface Item {
  id: number
  available_amount: number
  total_amount: number
  name: string
  description: string
  status: ItemStatus
}

export interface ItemCreateRequest {
  available_amount: number
  total_amount: number
  name: string
  description: string
  status: ItemStatus
}

export interface ItemPatchRequest {
  available_amount?: number
  total_amount?: number
  name?: string
  description?: string
  status?: ItemStatus
}

